galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/RunArtifactsDownloadRoute.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright contributors to the Galasa project
+ */
+package dev.galasa.framework.api.ras.internal;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.ByteChannel;
+import java.nio.file.FileSystem;
+import java.nio.file.OpenOption;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.attribute.FileAttribute;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.Gson;
+
+import dev.galasa.framework.IFileSystem;
+import dev.galasa.framework.spi.FrameworkException;
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.IRunResult;
+import dev.galasa.framework.spi.ResultArchiveStoreException;
+import dev.galasa.framework.spi.utils.GalasaGsonBuilder;
+
+import static dev.galasa.framework.api.ras.internal.ServletErrorMessage.*;
+
+public class RunArtifactsDownloadRoute extends RunsRoute {
+    
+    static final Gson gson = GalasaGsonBuilder.build();
+
+    public RunArtifactsDownloadRoute(IFileSystem fileSystem, IFramework framework) {
```

Comment:
Some comments about what this regex aims to match against would be good ? Are there unit tests for this, as regex changes often cause bugs...

Reply:
Might be better to ask each route "does this match something you can handle" rather than getting the superclass to do the regex ? Then you can easily unit test it.